kroxylicious-filters/kroxylicious-user-namespace/src/main/templates/UserNamespaceFilter.ftl

```diff
@@ -0,0 +1,354 @@
+                <#assign hashAsString = "">
+                <#list entityFields as key, value>
+                    <#assign append = "=" + key + ", ">
+                    <#assign hashAsString = hashAsString + append>
+                </#list>
+
+                // ${specName?trim} has fields ${hashAsString}
+                <#list entityFields.entities as entity>
+                    LOGGER.atDebug()
+                            .addArgument(context.sessionId())
+                            .addArgument(aid)
+                            .addArgument(${dataVar})
+                            .log("{} for {}: received ${specName?trim}: {}");
+                    <#assign getter="${entity.name?uncap_first}" setter="set${entity.name}" />
+                   // ${messageSpec.name} ${entity.name} ${entity.type} ${entity.entityType} ${entity.type.isArray?string('true', 'false')}
+                   if (shouldMap("${entity.entityType}") && inVersion(header.requestApiVersion(),
+                                ${specName?trim}_VERSIONS)) {
+                   <#if entity.type == 'string'>
+                        ${dataVar}.${setter}(map(aid, ${dataVar}.${getter}()));
+                   <#elseif entity.type == '[]string'>
+                        ${dataVar}.${setter}(${dataVar}.${getter}().stream().map(orig -> map(aid, orig)).toList());
+                   <#elseif entity.type.isArray>
+                        // Recurse
+                        // ${messageSpec.name} ${entity.name} ${entity.type} ${entity.entityType} ${entity.type.isArray?string('true', 'false')}
+                       <@applyNamespacePrefix entity.type entity.entityFields />
+                   </#if>
+                        LOGGER.atDebug()
+                                .addArgument(context.sessionId())
+                                .addArgument(aid)
+                                .addArgument(${dataVar})
+                                .log("{} for {}: result ${specName?trim}: {}");
+                    }
+                </#list>
+
+            <#--
+                <@dumpFoo childFoo />
+            -->
+            </#macro>
+
+
+            switch (apiKey) {
+<#list messageSpecs as messageSpec>
+    <#if messageSpec.type?lower_case == 'request' && messageSpec.entityFields.hasAtLeastOneEntityField>
+                case ${retrieveApiKey(messageSpec)} -> {
+                  <@applyNamespacePrefix messageSpec messageSpec.entityFields />
+                }
+    </#if>
+</#list>
+            }
+
+
+        });
+        return context.forwardRequest(header, request);
+
+    }
+
+    private boolean shouldMap(String entityType) {
+        try {
+            return config.resourceTypes().contains(UserNamespace.ResourceType.valueOf(entityType));
+        }
+        catch (IllegalArgumentException e) {
+            return false;
+        }
+    }
+
+    private static String map(String authId, String originalName) {
```

Comment:
This name leads to calling code like `stream.map(orig -> map(aid, orig))`. Two `maps` are one too many for my feeble brain. Something like `namespacedName` would be better, imho.